src/Symfony/Component/Form/FormFactory.php

```diff
@@ -60,9 +60,23 @@ public function createForProperty($class, $property, $data = null, array $option
      */
     public function createBuilder($type = 'form', $data = null, array $options = array())
     {
-        $name = $type instanceof FormTypeInterface || $type instanceof ResolvedFormTypeInterface
-            ? $type->getName()
-            : $type;
+        if (!$type instanceof FormTypeInterface && !$type instanceof ResolvedFormTypeInterface) {
+            $type = $this->registry->getType($type);
+        }
+
+        $typeName = $type->getName();
+
+        if (false === ($pos = strrpos($typeName, '\\'))) {
+            // No FQCN - leave unchanged for BC
+            $name = $typeName;
+        } else {
+            // FQCN - strip namespace and "Type" suffix
+            $name = strtolower(substr($typeName, $pos + 1));
+
+            if ('type' === substr($name, -4)) {
+                $name = substr($name, 0, -4);
```

Comment:
If the class name is just `Type` this would result in an empty name. Maybe only strip the type suffix if strlen > 4?